util/execdetails/execdetails.go

```diff
@@ -376,11 +394,82 @@ type BasicRuntimeStats struct {
 	rows int64
 }
 
-// GetActRows implements the RuntimeStats interface.
+// GetActRows return total rows of BasicRuntimeStats.
 func (e *BasicRuntimeStats) GetActRows() int64 {
 	return e.rows
 }
 
+// Clone implements the RuntimeStats interface.
+func (e *BasicRuntimeStats) Clone() RuntimeStats {
+	return &BasicRuntimeStats{
+		loop:    e.loop,
+		consume: e.consume,
+		rows:    e.rows,
+	}
+}
+
+// Merge implements the RuntimeStats interface.
+func (e *BasicRuntimeStats) Merge(rs RuntimeStats) {
+	tmp, ok := rs.(*BasicRuntimeStats)
+	if !ok {
+		return
+	}
+	e.loop += tmp.loop
+	e.consume += tmp.consume
+	e.rows += tmp.rows
+}
+
+// RootRuntimeStats is the executor runtime stats that combine with multiple runtime stats.
+type RootRuntimeStats struct {
+	basics   []*BasicRuntimeStats
+	groupRss [][]RuntimeStats
+}
+
+// GetActRows return total rows of RootRuntimeStats.
+func (e *RootRuntimeStats) GetActRows() int64 {
+	num := int64(0)
+	for _, basic := range e.basics {
+		num += basic.GetActRows()
+	}
+	return num
+}
+
+// String implements the RuntimeStats interface.
+func (e *RootRuntimeStats) String() string {
+	buf := bytes.NewBuffer(make([]byte, 0, 32))
+	if len(e.basics) > 0 {
+		if len(e.basics) == 1 {
```

Comment:
We can use a function to simplify the logic here.

Reply:
great, done.